sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/Conditions/ComparisonCondition.cs

```diff
@@ -41,15 +40,20 @@ public override string Stringify()
             conditionString.Append($"{Field} {QueryConstants.ComparisonOperators[Operator]} ");
 
             // check to see if the input is numeric value -- if not, we need single quotes around Value
-            bool isNumeric = int.TryParse(Value, out int numericValue);
-
-            if (!isNumeric)
+            if (typeof(T) == typeof(string))
```

Comment:
This looks better.

My other thought is that in SQL there are a few data types than can be treated as something other than strings, e.g. int/long, double/float, and bool. There might be others. I'm not sure. We could special case them, and then everything is is treated as a string (meaning we insert it between tick marks).